src/main/java/com/google/cloud/tools/app/module/DeleteAction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app.module;
+
+import com.google.cloud.tools.app.Action;
+import com.google.cloud.tools.app.GCloudExecutionException;
+import com.google.cloud.tools.app.Option;
+import com.google.cloud.tools.app.ProcessCaller;
+import com.google.cloud.tools.app.ProcessCaller.ProcessCallerFactory;
+import com.google.cloud.tools.app.ProcessCaller.Tool;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Deletes a version of one or more modules.
+ */
+public class DeleteAction extends Action {
+
+  private static ProcessCallerFactory processCallerFactory = ProcessCaller.getFactory();
+  private Collection<String> modules = UNSET_COLLECTION;
+  private String version = UNSET_STRING;
+  private String server = UNSET_STRING;
+
+  public static DeleteAction newDeleteAction() {
+    return new DeleteAction();
+  }
+
+  public DeleteAction setModules(Collection<String> modules) {
+    Preconditions.checkArgument(this.modules == UNSET_COLLECTION, "Modules can only be set once.");
+    this.modules = modules;
+    return this;
+  }
+
+  public DeleteAction setVersion(String version) {
+    Preconditions.checkArgument(this.version == UNSET_STRING, "Version can only be set once.");
+    this.version = version;
+    return this;
+  }
+
+  public DeleteAction setServer(String server) {
+    Preconditions.checkArgument(this.server == UNSET_STRING, "Server can only be set once.");
+    this.server = server;
+    return this;
+  }
+
+  @Override
+  public boolean execute() throws GCloudExecutionException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(version));
+
+    List<String> arguments = new ArrayList<>();
+    arguments.add("modules");
+    arguments.add("delete");
+    if (modules == null) {
+      arguments.add("default");
```

Comment:
When we discuss calling this from Maven, I think it makes sense for users to have one tool that does everything. In this case, you could do all kinds of management operations from Maven and wouldn't need any other tool. But I'm sure other folks will think this doesn't make sense in a build tool..

I opted for leaving modules optional so the API is lighter. e.g., if we weren't using gcloud CLI, no module means it's the default module. (Is that a safe assumption to make, or is that why gcloud makes users provide "default"?)
But I see an issue when gcloud changes "default" to something else, unlikely as that is.